libs/ml/include/ml/ops/divide.hpp

```diff
@@ -44,39 +45,81 @@ class Divide : public fetch::ml::Ops<T>
   virtual void Forward(VecTensorType const &inputs, ArrayType &output)
   {
     assert(inputs.size() == 2);
-    assert(inputs.at(0).get().shape() == inputs.at(1).get().shape());
     assert(inputs.at(0).get().shape() == output.shape());
 
-    fetch::math::Divide(inputs.at(0).get(), inputs.at(1).get(), output);
+    if ((inputs.at(0).get().shape() == inputs.at(1).get().shape()) ||
+        (inputs.at(1).get().size() > 1))
+    {  // array / array
+      fetch::math::Divide(inputs.at(0).get(), inputs.at(1).get(), output);
+    }
+    else
+    {  // array / scalar
+      fetch::math::Divide(inputs.at(0).get(), *(inputs.at(1).get().cbegin()), output);
+    }
   }
 
   /**
-   * f'(x)=(1/y)*err
-   * f'(y)=-(x/(y^2))*err
+   * f'(a)=(1/b)*err
+   * f'(b)=-(a/(b^2))*err
    */
   virtual std::vector<ArrayType> Backward(VecTensorType const &inputs,
                                           ArrayType const &    error_signal)
   {
     ArrayType return_signal_1(inputs.at(0).get().shape());
-    ArrayType return_signal_2(return_signal_1.shape());
+    ArrayType return_signal_2(inputs.at(1).get().shape());
+    return_signal_2.Fill(static_cast<DataType>(0));
 
     auto a_it   = inputs.at(0).get().cbegin();
     auto b_it   = inputs.at(1).get().cbegin();
     auto err_it = error_signal.cbegin();
     auto r_1_it = return_signal_1.begin();
     auto r_2_it = return_signal_2.begin();
-    while (a_it.is_valid())
-    {
-      *r_1_it = (*err_it) / (*b_it);
-      *r_2_it = ((*err_it) * (*a_it)) / ((*b_it) * (*b_it));
+    if (inputs.at(0).get().shape() == inputs.at(1).get().shape())
+    {  // array / array same shape
+      while (a_it.is_valid())
+      {
+        *r_1_it = (*err_it) / (*b_it);
+        *r_2_it = -((*err_it) * (*a_it)) / ((*b_it) * (*b_it));
 
-      ++a_it;
-      ++b_it;
-      ++err_it;
-      ++r_1_it;
-      ++r_2_it;
+        ++a_it;
+        ++b_it;
+        ++err_it;
+        ++r_1_it;
+        ++r_2_it;
+      }
     }
+    else if (inputs.at(1).get().size() == 1)
+    {  // array / scalar
+      while (a_it.is_valid())
+      {
+        *r_1_it = (*err_it) / (*b_it);
+        *r_2_it += -((*err_it) * (*a_it)) / ((*b_it) * (*b_it));
 
+        ++a_it;
+        ++err_it;
+        ++r_1_it;
+      }
+    }
+    else
+    {  // array / array different shape
+      // TODO () This implementation only works in limited scenarios, as it does not resolve the
```

Comment:
please raise an issue that Divide backwards broadcast needs to be implemented. You can paste this code snippet there if you think it would be helpful, and you must delete it from the codebase here.